WordPress/Sniffs/NamingConventions/PrefixAllGlobalsSniff.php

```diff
@@ -0,0 +1,503 @@
+					break;
+
+				case 'T_CONST':
+					// Constants in a class do not need to be prefixed.
+					if ( $this->phpcsFile->hasCondition( $stackPtr, array( T_CLASS, T_ANON_CLASS, T_INTERFACE, T_TRAIT ) ) === true ) {
+						return;
+					}
+
+					$constant_name_ptr = $this->phpcsFile->findNext( PHP_CodeSniffer_Tokens::$emptyTokens, ( $stackPtr + 1 ), null, true, null, true );
+					if ( false === $constant_name_ptr ) {
+						// Live coding.
+						return;
+					}
+
+					$item_name = $this->tokens[ $constant_name_ptr ]['content'];
+					$error_text = 'Global constants defined';
+					$error_code = 'NonPrefixedConstantFound';
+					break;
+
+				default:
+					// Left empty on purpose.
+					break;
+
+			} // End switch().
+
+			if ( empty( $item_name ) || $this->is_prefixed( $item_name ) === true ) {
+				return;
+			}
+
+			$this->phpcsFile->addError(
+				self::ERROR_MSG,
+				$stackPtr,
+				$error_code,
+				array(
+					$error_text,
+					$item_name,
+				)
+			);
+		} // End if().
+
+	} // End process_token().
+
+	/**
+	 * Check that defined global variables are prefixed.
+	 *
+	 * @since 0.12.0
+	 *
+	 * @param int $stackPtr The position of the current token in the stack.
+	 *
+	 * @return int|void Integer stack pointer to skip forward or void to continue
+	 *                  normal file processing.
+	 */
+	protected function process_variable_assignment( $stackPtr ) {
+
+		// We're only concerned with variables which are being defined.
+		if ( false === $this->is_assignment( $stackPtr ) ) {
+			return;
+		}
+
+		$variable_name = substr( $this->tokens[ $stackPtr ]['content'], 1 ); // Strip the dollar sign.
+
+		if ( 'GLOBALS' === $variable_name ) {
+			$array_open = $this->phpcsFile->findNext( PHP_CodeSniffer_Tokens::$emptyTokens, ( $stackPtr + 1 ), null, true, null, true );
+			if ( false === $array_open ) {
+				// Live coding.
+				return;
+			}
+
+			$array_key = $this->phpcsFile->findNext( PHP_CodeSniffer_Tokens::$emptyTokens, ( $array_open + 1 ), null, true, null, true );
+			if ( false === $array_key || T_CONSTANT_ENCAPSED_STRING !== $this->tokens[ $array_key ]['code'] ) {
+				// Key is not a string, nothing to do.
+				return;
+			}
+
+			$stackPtr      = $array_key;
+			$variable_name = $this->strip_quotes( $this->tokens[ $array_key ]['content'] );
+
+		} else {
+			// Local variables in a function do not need to be prefixed unless they are being imported.
+			if ( $this->phpcsFile->hasCondition( $stackPtr, array( T_FUNCTION, T_CLOSURE ) ) === true ) {
+				$condition = $this->phpcsFile->getCondition( $stackPtr, T_FUNCTION );
+				if ( false === $condition ) {
+					$condition = $this->phpcsFile->getCondition( $stackPtr, T_CLOSURE );
+				}
+
+				$has_global = $this->phpcsFile->findPrevious( T_GLOBAL, ( $stackPtr - 1 ), $this->tokens[ $condition ]['scope_opener'] );
+				if ( false === $has_global ) {
+					// No variable import happening.
+					return;
+				}
+
+				// Ok, this may be an imported global variable.
```

Comment:
> I though there was a sniff which was supposed to check this ?

Checked & there is, but turns out that sniff had some bugs in it when there was a mix of tabs & spaces on a line.

Fixed & pulled upstream: https://github.com/squizlabs/PHP_CodeSniffer/pull/1404, so hopefully these type of issues will soon be caught by the travis builds ;-)